eslint.config.mjs

```diff
@@ -0,0 +1,152 @@
+import { FlatCompat } from "@eslint/eslintrc";
+import eslint from '@eslint/js';
+import pluginLingui from 'eslint-plugin-lingui'
+import reactPlugin from 'eslint-plugin-react';
+import reactHooksPlugin from 'eslint-plugin-react-hooks';
+import reactCompilerPlugin from 'eslint-plugin-react-compiler';
+import reactNativeA11yPlugin from 'eslint-plugin-react-native-a11y';
+import simpleImportSort from "eslint-plugin-simple-import-sort";
+import path from "path";
+import tseslint from 'typescript-eslint';
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: eslint.configs.recommended,
+});
+
+export default tseslint.config(
+  eslint.configs.recommended,
+  ...tseslint.configs.recommended,
+  reactPlugin.configs.flat.recommended,
+  reactPlugin.configs.flat['jsx-runtime'],
```

Comment:
IMO, this was not extended in the legacy config, so it shouldn't be extended in the flat config for now, at least not in this PR.

Reply:
I mean the `jsx-runtime` one… it's causing `import React` being removed everywhere. Removing it could significantly scale down the diff of this PR.

Reply:
Totally understand. Yeah I was thinking of doing JSX first since I can technically do it with existing config. It's a fairly low risk change anyways.

Reply:
@haoqunjiang I have broken the JSX transform + eslint config into #6516 6516